File: archives/leetcode2/0232. Implement Queue using Stacks.py
class MyQueue(object):

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.stack=[]

    def push(self, x):
        """
        Push element x to the back of queue.
        :type x: int
        :rtype: void
        """
        tmp=[]
        while self.stack:
            tmp.append(self.stack.pop())
        self.stack.append(x)
        while tmp:
            self.stack.append(tmp.pop())

    def pop(self):
        """
        Removes the element from in front of queue and returns that element.
        :rtype: int
        """
        return self.stack.pop()

    def peek(self):
        """
        Get the front element.
        :rtype: int
        """
        return self.stack[-1]

    def empty(self):
        """
        Returns whether the queue is empty.
        :rtype: bool
        """
        return self.stack==[]
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: archives/C#/0232. Implement Queue using Stacks.cs
public class MyQueue {

    Stack<int> myQueue= new Stack<int>();
    Stack<int> tmpQueue= new Stack<int>();
    /** Initialize your data structure here. */
    public MyQueue() {
        
    }
    
    /** Push element x to the back of queue. */
    public void Push(int x) {
        if(myQueue.Count()==0){
            myQueue.Push(x);
        }
        else{
            while(myQueue.Count()!=0){
                int i=myQueue.Pop();
                tmpQueue.Push(i);
            }
            myQueue.Push(x);
            while(tmpQueue.Count()!=0){
                int i=tmpQueue.Pop();
                myQueue.Push(i);
            }            
        }
    }
    
    /** Removes the element from in front of queue and returns that element. */
    public int Pop() {
        return myQueue.Pop();
    }
    
    /** Get the front element. */
    public int Peek() {
        return myQueue.Peek();
    }
    
    /** Returns whether the queue is empty. */
    public bool Empty() {
        return myQueue.Count()==0;
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.Push(x);
 * int param_2 = obj.Pop();
 * int param_3 = obj.Peek();
 * bool param_4 = obj.Empty();
 */
